modules/kafka/kafka-dest-worker.c

```diff
@@ -86,8 +82,8 @@ _publish_message(KafkaDestWorker *self, LogMessage *msg)
 
   /* we passed the allocated buffers to rdkafka, which will eventually free them */
   g_string_steal(message);
-  if (key->str)
-    g_string_steal(key);
+  /* key is not freed by rdkafka */
+  g_string_set_size(key, 0);
```

Comment:
`rd_kafka_produce()` copies the key for itself, I checked. :+1:

Reply:
Could you remove the comment, please? I think it's not worth mentioning the opposite.

Reply:
`log_template_format()` truncates the string, you can actually remove this line. I'd also use `self->key` directly, removing the local key variable.

Reply:
Done! :)